### ISSUE TITLE
Learning rate adjustments during train

### ISSUE THREAD

**Issue by larjung** [InterDigitalInc/CompressAI#8]
Hi,
Thanks for your great project!

I am trying to figure out the exact learning rate adjustment regime that you are using to reproduce the training of bmshj2018, mbt2018 etc. I couldn't found any indication in the code (e.g. in train.py), however you did mention in the [documentation](interdigitalinc.github.io/CompressAI/zoo.html#models) :

> _Models were trained with a batch size of 16 or 32, and an initial learning rate of 1e-4 for approximately 1-2M steps. **The learning rate is then divided by 2 when the evaluation loss reaches a plateau (we use a patience of 20 epochs)**._

From this I infer that you're probably using pytorch's **ReduceLROnPlateau** with patience 20 and factor 0.5.
My questions are:
- Did you apply ReduceLROnPlateau only on the optimizer of the main loss, or also for the **auxiliary loss** that is being optimized during training as well? 
- If ReduceLROnPlateau is applied also on the auxiliary loss (that is initialized with LR of 1e-3), then what are its parameters? Are they identical to the main loss?
- What about the other parameters for ReduceLROnPlateau (e.g. threshold, cooldown, min_lr, etc.)? If there are other non-default values that you used, could you please specify?

Another question on a slightly different topic: When training for MS-SSIM metric, have you trained the models from scratch, or finetuned from pre-trained models optimized for MSE (some papers mention they are using the latter option to save time).

All the best,
Danny

**Comment by jbegaint** [comment]
Hi larjung,

Thanks for pointing out those missing details, I'll update the doc soon!

The `train.py` is just meant to be an example script. Internally we use a more complex training platform but we try to provide details in the documentation. To answer your questions:

- ReduceLROnPlateau is only applied on the main loss
- We use a patience of 20 and a factor of 0.5, all the other parameters are kept to the default pytorch values.

It's definitely possible that this simple LR scheduling is not optimal and a better one could be found to train the networks faster. I performed some experiments (warm up, cosine annealing, etc...) but could not find a better alternative.  

And yes, we did train the MS-SSIM networks from pre-trained MSE networks. Training is faster and more stable. We used a learning rate of 1e-5 for both optimizers.

Let me know if you have more questions!
